File: test/c++11/intel_x64/cpuid.cpp
#include <pal/instruction/cpuid.h>
#include <pal/cpuid/leaf_01_eax.h>

void test_cpuid_compile(void)
{
    // Inputs
    uint32_t leaf{0x1};
    uint32_t subleaf{0x0};

    // Outputs
    uint32_t eax{0};
    uint32_t ebx{0};
    uint32_t ecx{0};
    uint32_t edx{0};

    std::tie(eax, ebx, ecx, edx) = pal::execute_cpuid(leaf, subleaf);

    pal::leaf_01_eax::print(eax);
    uint32_t family_id = pal::leaf_01_eax::family_id::get(eax);
}

File: test/rust/intel/src/cpuid.rs
use pal;

pub unsafe fn test_cpuid_compile()
{
    let leaf = 0x1;
    let subleaf = 0x0;

    let (eax, ebx, ecx, edx) = pal::instruction::execute_cpuid(leaf, subleaf);

    #[cfg(feature = "pal/enable_printers")]
    {
    pal::cpuid::leaf_01_eax::print_from_value(eax);
    pal::cpuid::leaf_01_ebx::print_from_value(ebx);
    pal::cpuid::leaf_01_ecx::print_from_value(ecx);
    pal::cpuid::leaf_01_edx::print_from_value(edx);
    }

    #[cfg(not(feature = "pal/enable_printers"))]
    let _ = (eax, ebx, ecx, edx);
}
